agent/health/health.go

```diff
@@ -99,7 +99,7 @@ func (r Probes) GetFailed() []*pb.Probe {
 func (r Probes) Status() pb.NodeStatus_Type {
 	result := pb.NodeStatus_Running
 	for _, probe := range r {
-		if probe.Status == pb.Probe_Failed {
+		if probe.Status == pb.Probe_Failed && probe.Severity != pb.Probe_Warning {
```

Comment:
Nit: not sure I'm a huge fan of writing the logic as a `!=`. If new statuses get added to the protobuf, they may interact here in an unpredictable way. Right now I believe this will cause anyone creating a probe with a Status failed and severity of None to return a status of degraded? Or any new severity to also show as degraded?

Reply:
Yeah, that's actually how it works right now (or rather, has worked until now). If you look at the triggered (failed) probes - none of them actually have severity set. I'm not sure why that is, I guess for legacy reasons, and I think we'll need to go and make sure each probe has an appropriate severity set eventually.